lib/NewFangle/Config.pm

```diff
@@ -67,7 +67,7 @@ used.
     my($xsub, $class, %config) = @_;
     my $app_name    = delete $config{app_name}    // $ENV{NEWRELIC_APP_NAME}    // 'AppName';
     my $license_key = delete $config{license_key} // $ENV{NEWRELIC_LICENSE_KEY} // '';
-    my $config = $xsub->($app_name, $license_key);
+    my $config = $xsub->($app_name, $license_key) // die("create_app_config failed");
```

Comment:
We should use carp here instead of die.

Since `$app_name` can't be `undef` / `NULL`, can we assume the error is in the license key and have the message be something like "Error creating NetFangle::Config, bad license key"

Reply:
I was aiming for better diagnostics because it took me a while to figure out what the cryptic error message (`Not an object`) actually meant. I like the message you suggested. That's a good idea :+1: